akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -4,22 +4,21 @@
 package akka.stream.impl
 
 import java.util.concurrent.atomic.AtomicReference
-
 import akka.actor.{ ActorRef, Props }
 import akka.stream.impl.StreamLayout.Module
 import akka.stream.scaladsl.OperationAttributes
 import akka.stream.{ Inlet, Shape, SinkShape }
 import org.reactivestreams.{ Publisher, Subscriber, Subscription }
-
 import scala.annotation.unchecked.uncheckedVariance
 import scala.concurrent.{ Future, Promise }
+import akka.stream.impl.ActorFlowMaterializerImpl.CreateContext
 
 /**
  * INTERNAL API
  */
 private[akka] abstract class SinkModule[-In, Mat](val shape: SinkShape[In]) extends Module {
 
-  def create(materializer: ActorFlowMaterializerImpl, flowName: String): (Subscriber[In] @uncheckedVariance, Mat)
+  def create(context: CreateContext): (Subscriber[In] @uncheckedVariance, Mat)
```

Comment:
yes, I was thinking how far I should go, but stayed here. As far as I can see we don't really support custom sources and sinks anyway, or did I miss anything?

I can make CreateContext independent of actor if you like, but then I think the actorOf method should still be on ActorFlowMaterializer, taking the CreateContext and the Props as parameters.
The source/sink have to downcast to ActorFlowMaterializer when needed.

I think that would be better than CreateContext and ActorCreateContext.

I'll give it a quick try.